### PR TITLE
Edit comment

### PR DESCRIPTION
**Pull request by joassouza**
This PR allows to edit a comment or a comment reply.

### CODE REVIEW

File: commentManager.js [hunk 1]
```diff
@@ -200,3 +200,40 @@ exports.changeAcceptedState = function(padId, commentId, state, callback){
...
+    db.get(prefix + padId, function(err, comments){
+      if(ERR(err, callback)) return;
+
+      //add the entry for this pad
+      var comment = comments[commentId];
+      comment.text = commentText;
+      comments[commentId] = comment;
+
+      //save the new element back
+      db.set(prefix + padId, comments);
```
Comment (lpagliari) [inline commentManager.js]: `//add the entry for this pad`: we're not adding, but updating the entry, right?

`//save the new element back`: it's not a new element

Comment (lpagliari) [inline commentManager.js]: One suggestion, if you want to simplify the code:

```
      var comment = comments[commentId];
      comment.text = commentText;
      comments[commentId] = comment;
```

could be
```
     comments[commentId].text = commentText;
```

File: index.js [hunk 2]
```diff
@@ -78,6 +78,20 @@ exports.socketio = function (hook_name, args, cb){
       });
     });
 
+    socket.on('commentTextUpdated', function(data, callback) {
```
Comment (lpagliari) [inline index.js]: Following the standard of other events, this one should be `updateCommentText`, not `commentTextUpdated`

File: static/js/index.js [hunk 3]
```diff
@@ -334,6 +408,25 @@ ep_comments.prototype.init = function(){
...
+  if (!hasEditForm) {
+    // get text from comment
+    var commentTextValue = $commentBox.find('.comment-text').text();
+
+    // add a form to edit the field
+    var data = {};
+    data.text = commentTextValue;
+    var content = $("#editCommentTemplate").tmpl(data);
+
+    // localize comment reply
```
Comment (lpagliari) [inline static/js/index.js]: I believe `content` is not a comment reply, right?

File: commentManager.js [hunk 4]
```diff
@@ -200,3 +200,40 @@ exports.changeAcceptedState = function(padId, commentId, state, callback){
...
+    });
+  };
+
+  // If we're dealing with comment replies we need to a different query
+  var prefix = "comments:";
+  if(commentId.substring(0,7) === "c-reply"){
+    prefix = "comment-replies:";
+  }
+
+  var commentTextIsEmpty = commentText.length > 0;
```
Comment (joassouza) [inline commentManager.js]: yep

File: commentManager.js (hunk 1 above)
Comment (joassouza) [inline commentManager.js]: thanks for the suggestion

**Comment by joassouza** [comment]
Thanks for the code review @lpagliari. The last commit implements the suggestions.

**Comment by kiranpyati** [comment]
Nice feature, thank you. But I see one issue here that I can edit other users comments as well.
Expected: Only owner of the comment should edit comment.

**Pull request merged**